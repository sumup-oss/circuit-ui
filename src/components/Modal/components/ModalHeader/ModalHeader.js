import React from 'react';
import PropTypes from 'prop-types';
import { withTheme } from 'emotion-theming';

import { CardHeader } from '../../../Card';
import Heading from '../../../Heading';

const ModalHeader = ({ title, onClose }) => (
  <CardHeader onClose={onClose}>
    {title && (
      <Heading size={Heading.KILO} noMargin>
        {title}
      </Heading>
    )}
  </CardHeader>
);

ModalHeader.propTypes = {
  onClose: PropTypes.func,
  title: PropTypes.string.isRequired
};

ModalHeader.defaultProps = {
  onClose: null
};

/**
 * @component
 */
export default withTheme(ModalHeader);
