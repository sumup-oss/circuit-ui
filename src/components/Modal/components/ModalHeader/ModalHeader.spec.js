import React from 'react';

import ModalHeader from '.';

describe('ModalHeader', () => {
  /**
   * Style tests.
   */
  it('should render with default styles', () => {
    const actual = create(<ModalHeader />);
    expect(actual).toMatchSnapshot();
  });

  /**
   * Accessibility tests.
   */
  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(<ModalHeader />);
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });
});
