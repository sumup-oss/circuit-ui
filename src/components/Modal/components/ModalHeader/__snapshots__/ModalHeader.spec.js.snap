// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader should render with default styles 1`] = `
.circuit-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 24px;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<header
  class="circuit-0 circuit-1"
/>
`;
